src/bin/pg_autoctl/pgsql.c

```diff
@@ -151,9 +156,44 @@ pgsql_open_connection(PGSQL *pgsql)
 	/* Check to see that the backend connection was successfully made */
 	if (PQstatus(connection) != CONNECTION_OK)
 	{
-		log_error("Connection to database failed: %s", PQerrorMessage(connection));
-		pgsql_finish(pgsql);
-		return NULL;
+		/* We failed to connect to Postgres. When connecting to a local node,
+		 * we are going to handle the situation by probing the postmasted.pid
+		 * file. When connecting to a remove node (monitor or coordinator), we
+		 * want to implement a retry loop: it might be a transient failure,
+		 * such as when the remote node is not ready yet.
+		 */
+		switch (pgsql->connectionType)
+		{
+			case PGSQL_CONN_LOCAL:
+			{
+				log_error("Connection to database failed: %s",
+						  PQerrorMessage(connection));
+				pgsql_finish(pgsql);
+				return NULL;
+			}
+
+			case PGSQL_CONN_MONITOR:
+			case PGSQL_CONN_COORDINATOR:
```

Comment:
Is this intentional (waiting until coordinator is available) ?

Reply:
Yes it is. The coordinator could be the last node provisioned and we want to handle that situation just fine.